### DISCUSSION TITLE
Do I need to expose "security definer" Functions in Row Level Security Policies?

### DISCUSSION THREAD

**Discussion by kiwicopple** [supabase/supabase#16784] [Troubleshooting]
_This is a copy of a troubleshooting article on Supabase's docs site. It may be missing some details from the original. View the [original article](https://supabase.com/docs/guides/troubleshooting/do-i-need-to-expose-security-definer-functions-in-row-level-security-policies-iI0uOw)._

PostgREST supports 2 config parameters:

*   Exposed Schemas
*   Extra Search Path

![image](/docs/img/troubleshooting/d756aeb0-515f-425d-b737-75a935935b73.png)

You do not need to add your "security definer" Functions to either of these if you are using them in your Policies.

PostgREST doesn’t need to know about this function on extra search path or exposed schemas, as long as you explicitly use the schema inside RLS (e.g.: `security.rls_func`).

**Comment by IVFDADDIES** [comment]
SELECT
  n.nspname AS schema,
  c.relname AS view_name,
  pg_get_userbyid(c.relowner) AS owner,
  c.relsecurity,
  c.relkind,
  pg_catalog.obj_description(c.oid, 'pg_class') AS comment,
  pg_catalog.pg_get_viewdef(c.oid, true) AS view_definition
FROM
  pg_class c
  JOIN pg_namespace n ON n.oid = c.relnamespace
WHERE
  c.relkind = 'v' -- only views
  AND c.relrowsecurity IS NOT TRUE -- RLS not enabled
  AND c.relsecurity = 1; -- SECURITY DEFINER
